libsolidity/interface/CompilerStack.cpp

```diff
@@ -1548,6 +1549,9 @@ string CompilerStack::createMetadata(Contract const& _contract, bool _forIR) con
 	if (m_revertStrings != RevertStrings::Default)
 		meta["settings"]["debug"]["revertStrings"] = revertStringsToString(m_revertStrings);
 
+	if (m_metadataFormat != MetadataFormat::NoMetadata)
```

Comment:
```suggestion
	if (m_metadataFormat != CompilerStack::defaultMetadataFormat())
```

Reply:
Ah, good catch. Will change to `==MetadataFormat::NoMetadata` since alex prefered it over `default`

Reply:
Just fixed.

Reply:
I'd actually oppose this, as it makes it way harder to understand the intent.

Reply:
Isn't the intent to skip fields that have default values?

Reply:
Yeah, that's right. Alex had a preference to using `NoMetadata` over `default`.

Reply:
> Isn't the intent to skip fields that have default values?

i think this is a confusing piece of code, as on the non-default case we include the an unrelated field with an unrelated value (compared to what the "default" says").